File: bTreeLevelOrder.cpp
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> ans;
        queue<TreeNode*> q;
        if(root == NULL){
            return ans;
        }
        q.push(root);
        while(q.empty()==false)
        {
            int c;
            c=q.size();
            vector<int> v;
            while(c--){
                v.push_back(q.front()->val);
                if(q.front()->left!=NULL)
                    q.push(q.front()->left);
                if(q.front()->right!=NULL)
                    q.push(q.front()->right);
                q.pop();
            }
            ans.push_back(v);
        }
        reverse(ans.begin(),ans.end());
        return ans;
    }
};

File: bTreeLevelOrder.py
from collections import deque
class Solution:
    def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:
        ans=[]
        if not root:
            return ans
        r=deque([root])
        while r:
            v=[]
            for i in range(len(r)):
                n=r.popleft()
                v.append(n.val)
                if n.left:
                    r.append(n.left)
                if n.right:
                    r.append(n.right)
            ans.append(v)
        return ans[::-1]
